lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -965,19 +1056,27 @@ def alerts(self) -> dict:
             if not alert_rules:
                 continue
 
+            identifier = None
             try:
                 scrape_metadata = json.loads(relation.data[relation.app]["scrape_metadata"])
-                topology = JujuTopology.from_relation_data(scrape_metadata)
-
-                alerts[topology.identifier] = alert_rules
-
+                identifier = ProviderTopology.from_relation_data(scrape_metadata).identifier
+                alerts[identifier] = alert_rules
             except KeyError as e:
-                logger.error(
-                    "Relation %s has invalid data : %s",
+                logger.warning(
+                    "Relation %s has no 'scrape_metadata': %s",
                     relation.id,
                     e,
                 )
 
+                # Construct an ID based on what's in the alert rules
+                labels = next(alert_rules["groups"])["labels"]
```

Comment:
- Is it possible that there are multiple groups in the alert rules ? I think so.
- Is it possible that the labels are different for different groups ? I am not sure
- In view of the two questions above is it OK to pick the labels of the first group. I am inclined to believe so but would like to hear from others.

Reply:
So in view of @simskij responses seems like we may need to loop over all groups here and create a separate topology identifier for each group.